rest/model/codegen.go

```diff
@@ -63,84 +63,113 @@ type modelConversionInfo struct {
 // ModelMapping maps schema type names to their respective DB model
 type ModelMapping map[string]string
 
-func Codegen(schema string, config ModelMapping) ([]byte, error) {
+func Codegen(schema string, config ModelMapping) ([]byte, []byte, error) {
 	source := ast.Source{
 		Input: schema,
 	}
 	fileTemplate, err := getTemplate(fileTemplatePath)
 	if err != nil {
-		return nil, errors.Wrap(err, "error loading file template")
+		return nil, nil, errors.Wrap(err, "error loading file template")
 	}
 	structTemplate, err := getTemplate(structTemplatePath)
 	if err != nil {
-		return nil, errors.Wrap(err, "error loading struct template")
+		return nil, nil, errors.Wrap(err, "error loading struct template")
 	}
 	fieldTemplate, err := getTemplate(fieldTemplatePath)
 	if err != nil {
-		return nil, errors.Wrap(err, "error loading field template")
+		return nil, nil, errors.Wrap(err, "error loading field template")
 	}
 	parsedAst, parseErr := gqlparser.LoadSchema(&source)
 	if parseErr != nil {
-		return nil, errors.Wrap(parseErr, "error parsing schema")
+		return nil, nil, errors.Wrap(parseErr, "error parsing schema")
 	}
 	catcher := grip.NewBasicCatcher()
 	structs := ""
-	conversionCode := ""
+	serviceCode := ""
+	generatedConversions := map[string]string{}
 
 	typeNames := []string{}
 	for key := range parsedAst.Types {
 		typeNames = append(typeNames, key)
 	}
 	sort.Strings(typeNames)
 	for _, typeName := range typeNames {
-		dbModel, shouldConvert := config[typeName]
-		if !shouldConvert {
+		gqlType := parsedAst.Types[typeName]
+		if gqlType == nil {
+			catcher.Errorf("unable to find type '%s'", typeName)
 			continue
 		}
-		gqlType := parsedAst.Types[typeName]
 		if gqlType.BuiltIn {
 			continue
 		}
 		if gqlType.Kind == ast.Scalar {
 			continue
 		}
-		fields := ""
-		extractedFields := extractedFields{}
-		for _, field := range gqlType.Fields {
-			fieldInfo := getFieldInfo(field)
-			extractedFields[field.Name] = fieldInfo
-			var fieldData string
-			fieldData, err = output(fieldTemplate, fieldInfo)
-			if err != nil {
-				catcher.Add(err)
-				continue
-			}
-			fields += fieldData
+		dbModel, shouldConvert := config[typeName]
```

Comment:
why did you move this to be after gqlType things? If it's not to catch the error at line 99 then you may as well short circuit

Reply:
haha well if you do find a good reason in future tickets let's drop a commend so we remember what's going on